Can not run maven flyway in Kitodo-DataManagement module

**Describe the bug**
Running the validation or migration of FlyWay through `mvn flyway:validate -Pflyway` in the Kitodo-DataManagement module results in the error

```
[Kitodo-DataManagement/] $ mvn flyway:validate -Pflyway
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< org.kitodo:kitodo-data-management >------------------
[INFO] Building Kitodo - Data Management 3.7.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.783 s
[INFO] Finished at: 2024-07-29T08:32:44+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project kitodo-data-management: Could not resolve dependencies for project org.kitodo:kitodo-data-management:jar:3.7.0-SNAPSHOT: Failed to collect dependencies at org.kitodo:kitodo-api:jar:3.7.0-SNAPSHOT: Failed to read artifact descriptor for org.kitodo:kitodo-api:jar:3.7.0-SNAPSHOT: org.kitodo:kitodo-production:pom:${revision} was not found in https://repo1.maven.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

The error message points to  wrong artifact descriptor `${revision}` which is not available. This was introduced by PR [#6127](https://github.com/kitodo/kitodo-production/pull/6127)


**To Reproduce**
1. Change into the Kitodo-DataManagement module
2. run `mvn flyway:validate -Pflyway`or `mvn flyway:migrate -Pflyway` 
3. See the error message


**Expected behavior**
Running the maven flyway for validation or migration in the Kitodo-DataManagement module should always be possible

**Release**
master branch (sha ee97b6883e850d760947a26dc73c4a26806bb63e)

I tried it on a clean git master with reverted commit 144b93fca098e86f7641bfe6120213568a6cbf68 and get a different error:
```
[ERROR] Failed to execute goal on project kitodo-data-management: Could not resolve dependencies for project org.kitodo:kitodo-data-management:jar:3.7.0-SNAPSHOT
[ERROR] dependency: org.kitodo:kitodo-api:jar:3.7.0-SNAPSHOT (provided)
[ERROR]   Could not find artifact org.kitodo:kitodo-api:jar:3.7.0-SNAPSHOT
```

> I tried it on a clean git master with reverted commit https://github.com/kitodo/kitodo-production/commit/144b93fca098e86f7641bfe6120213568a6cbf68 and get a different error:

I did not get your error - instead the flyway validation / migration did work from the Kitodo-DataManagement module:

```
[Kitodo-DataManagement] $ mvn flyway:validate -Pflyway
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< org.kitodo:kitodo-data-management >------------------
[INFO] Building Kitodo - Data Management 3.7.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- flyway-maven-plugin:8.4.4:validate (default-cli) @ kitodo-data-management ---
[INFO] Flyway Community Edition 8.4.4 by Redgate
[INFO] Database: jdbc:mariadb://localhost/kitodo_maven_37x (MariaDB 10.11)
[WARNING] Flyway upgrade recommended: MariaDB 10.11 is newer than this version of Flyway and support has not been tested. The latest supported version of MariaDB is 10.6.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.033 s
[INFO] Finished at: 2024-07-29T11:24:41+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:8.4.4:validate (default-cli) on project kitodo-data-management: org.flywaydb.core.api.exception.FlywayValidateException: Validate failed: Migrations have failed validation
[ERROR] Detected resolved migration not applied to database: 2.130. To fix this error, either run migrate, or set -ignorePendingMigrations=true.
[ERROR] Detected resolved migration not applied to database: 2.131. To fix this error, either run migrate, or set -ignorePendingMigrations=true.
[ERROR] Need more flexibility with validation rules? Learn more: https://rd.gt/3AbJUZE
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

The mentioned errors here are excepted as my local database was not already migrated.


> mvn flyway:validate -Pflyway and mvn flyway:migrate -Pflyway work with git master when they are started from the root directory.

Not for me:

```
[kitodo-production] $ mvn flyway:validate -Pflyway
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Kitodo Production                                                  [pom]
[INFO] Kitodo - API                                                       [jar]
[INFO] Kitodo - Data Management                                           [jar]
[INFO] Kitodo - Query URL Import                                          [jar]
[INFO] Kitodo - Command                                                   [jar]
[INFO] Kitodo - File Management                                           [jar]
[INFO] Kitodo - Data Format                                               [jar]
[INFO] Kitodo - Data Editor                                               [jar]
[INFO] Kitodo - Image Management                                          [jar]
[INFO] Kitodo - Long Term Preservation Validation                         [jar]
[INFO] Kitodo - Persistent Identifier                                     [jar]
[INFO] Kitodo - Docket                                                    [jar]
[INFO] Kitodo - XML SchemaConverter                                       [jar]
[INFO] Kitodo - Validation                                                [jar]
[INFO] Kitodo - Core                                                      [war]
[INFO] 
[INFO] --------------------< org.kitodo:kitodo-production >--------------------
[INFO] Building Kitodo Production 3.7.0-SNAPSHOT                         [1/15]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- flyway-maven-plugin:8.4.4:validate (default-cli) @ kitodo-production ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Kitodo Production 3.7.0-SNAPSHOT:
[INFO] 
[INFO] Kitodo Production .................................. FAILURE [  0.422 s]
[INFO] Kitodo - API ....................................... SKIPPED
[INFO] Kitodo - Data Management ........................... SKIPPED
[INFO] Kitodo - Query URL Import .......................... SKIPPED
[INFO] Kitodo - Command ................................... SKIPPED
[INFO] Kitodo - File Management ........................... SKIPPED
[INFO] Kitodo - Data Format ............................... SKIPPED
[INFO] Kitodo - Data Editor ............................... SKIPPED
[INFO] Kitodo - Image Management .......................... SKIPPED
[INFO] Kitodo - Long Term Preservation Validation ......... SKIPPED
[INFO] Kitodo - Persistent Identifier ..................... SKIPPED
[INFO] Kitodo - Docket .................................... SKIPPED
[INFO] Kitodo - XML SchemaConverter ....................... SKIPPED
[INFO] Kitodo - Validation ................................ SKIPPED
[INFO] Kitodo - Core ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.763 s
[INFO] Finished at: 2024-07-29T10:29:04+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:8.4.4:validate (default-cli) on project kitodo-production: org.flywaydb.core.api.FlywayException: No database found to handle jdbc:mariadb://localhost/kitodo_maven?useSSL=false&amp;serverTimezone=Europe/Berlin -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Running the flyway migration from the source root directory did not work in the past and there was no changes which made running from the source root directory working.

> Adding -f ../pom.xml helps (although this should not be required).

No, this did not help. I got with this a similar error message like running on the source root directory:

```
[Kitodo-Datamanagement] $ mvn flyway:validate -Pflyway -f ../pom.xml
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Kitodo Production                                                  [pom]
[INFO] Kitodo - API                                                       [jar]
[INFO] Kitodo - Data Management                                           [jar]
[INFO] Kitodo - Query URL Import                                          [jar]
[INFO] Kitodo - Command                                                   [jar]
[INFO] Kitodo - File Management                                           [jar]
[INFO] Kitodo - Data Format                                               [jar]
[INFO] Kitodo - Data Editor                                               [jar]
[INFO] Kitodo - Image Management                                          [jar]
[INFO] Kitodo - Long Term Preservation Validation                         [jar]
[INFO] Kitodo - Persistent Identifier                                     [jar]
[INFO] Kitodo - Docket                                                    [jar]
[INFO] Kitodo - XML SchemaConverter                                       [jar]
[INFO] Kitodo - Validation                                                [jar]
[INFO] Kitodo - Core                                                      [war]
[INFO] 
[INFO] --------------------< org.kitodo:kitodo-production >--------------------
[INFO] Building Kitodo Production 3.7.0-SNAPSHOT                         [1/15]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- flyway-maven-plugin:8.4.4:validate (default-cli) @ kitodo-production ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Kitodo Production 3.7.0-SNAPSHOT:
[INFO] 
[INFO] Kitodo Production .................................. FAILURE [  0.781 s]
[INFO] Kitodo - API ....................................... SKIPPED
[INFO] Kitodo - Data Management ........................... SKIPPED
[INFO] Kitodo - Query URL Import .......................... SKIPPED
[INFO] Kitodo - Command ................................... SKIPPED
[INFO] Kitodo - File Management ........................... SKIPPED
[INFO] Kitodo - Data Format ............................... SKIPPED
[INFO] Kitodo - Data Editor ............................... SKIPPED
[INFO] Kitodo - Image Management .......................... SKIPPED
[INFO] Kitodo - Long Term Preservation Validation ......... SKIPPED
[INFO] Kitodo - Persistent Identifier ..................... SKIPPED
[INFO] Kitodo - Docket .................................... SKIPPED
[INFO] Kitodo - XML SchemaConverter ....................... SKIPPED
[INFO] Kitodo - Validation ................................ SKIPPED
[INFO] Kitodo - Core ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.380 s
[INFO] Finished at: 2024-07-29T11:30:53+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:8.4.4:validate (default-cli) on project kitodo-production: org.flywaydb.core.api.FlywayException: No database found to handle jdbc:mysql://localhost/kitodo?useSSL=false -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Only working way for me is to revert the changes for PR #6127.

Okay, here is the log 

```
[kitodo-production] $ mvn flyway:validate -Pflyway
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Kitodo Production                                                  [pom]
[INFO] Kitodo - API                                                       [jar]
[INFO] Kitodo - Data Management                                           [jar]
[INFO] Kitodo - Query URL Import                                          [jar]
[INFO] Kitodo - Command                                                   [jar]
[INFO] Kitodo - File Management                                           [jar]
[INFO] Kitodo - Data Format                                               [jar]
[INFO] Kitodo - Data Editor                                               [jar]
[INFO] Kitodo - Image Management                                          [jar]
[INFO] Kitodo - Long Term Preservation Validation                         [jar]
[INFO] Kitodo - Persistent Identifier                                     [jar]
[INFO] Kitodo - Docket                                                    [jar]
[INFO] Kitodo - XML SchemaConverter                                       [jar]
[INFO] Kitodo - Validation                                                [jar]
[INFO] Kitodo - Core                                                      [war]
[INFO] 
[INFO] --------------------< org.kitodo:kitodo-production >--------------------
[INFO] Building Kitodo Production 3.7.0-SNAPSHOT                         [1/15]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- flyway-maven-plugin:8.4.4:validate (default-cli) @ kitodo-production ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Kitodo Production 3.7.0-SNAPSHOT:
[INFO] 
[INFO] Kitodo Production .................................. FAILURE [  0.239 s]
[INFO] Kitodo - API ....................................... SKIPPED
[INFO] Kitodo - Data Management ........................... SKIPPED
[INFO] Kitodo - Query URL Import .......................... SKIPPED
[INFO] Kitodo - Command ................................... SKIPPED
[INFO] Kitodo - File Management ........................... SKIPPED
[INFO] Kitodo - Data Format ............................... SKIPPED
[INFO] Kitodo - Data Editor ............................... SKIPPED
[INFO] Kitodo - Image Management .......................... SKIPPED
[INFO] Kitodo - Long Term Preservation Validation ......... SKIPPED
[INFO] Kitodo - Persistent Identifier ..................... SKIPPED
[INFO] Kitodo - Docket .................................... SKIPPED
[INFO] Kitodo - XML SchemaConverter ....................... SKIPPED
[INFO] Kitodo - Validation ................................ SKIPPED
[INFO] Kitodo - Core ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.461 s
[INFO] Finished at: 2024-07-29T13:01:40+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:8.4.4:validate (default-cli) on project kitodo-production: org.flywaydb.core.api.FlywayException: No database found to handle jdbc:mariadb://localhost/kitodo_maven_37x -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

No difference.

--- a/Kitodo-API/pom.xml
+++ b/Kitodo-API/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - API</name>

--- a/Kitodo-Command/pom.xml
+++ b/Kitodo-Command/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Command</name>

--- a/Kitodo-DataEditor/pom.xml
+++ b/Kitodo-DataEditor/pom.xml
@@ -17,7 +17,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Data Editor</name>

--- a/Kitodo-DataFormat/pom.xml
+++ b/Kitodo-DataFormat/pom.xml
@@ -17,7 +17,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Data Format</name>

--- a/Kitodo-DataManagement/pom.xml
+++ b/Kitodo-DataManagement/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Data Management</name>

--- a/Kitodo-Docket/pom.xml
+++ b/Kitodo-Docket/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Docket</name>

--- a/Kitodo-FileManagement/pom.xml
+++ b/Kitodo-FileManagement/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - File Management</name>

--- a/Kitodo-ImageManagement/pom.xml
+++ b/Kitodo-ImageManagement/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Image Management</name>

--- a/Kitodo-LongTermPreservationValidation/pom.xml
+++ b/Kitodo-LongTermPreservationValidation/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Long Term Preservation Validation</name>

--- a/Kitodo-PersistentIdentifier/pom.xml
+++ b/Kitodo-PersistentIdentifier/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Persistent Identifier</name>

--- a/Kitodo-Query-URL-Import/pom.xml
+++ b/Kitodo-Query-URL-Import/pom.xml
@@ -17,7 +17,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Query URL Import</name>

--- a/Kitodo-Validation/pom.xml
+++ b/Kitodo-Validation/pom.xml
@@ -19,7 +19,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Validation</name>

--- a/Kitodo-XML-SchemaConverter/pom.xml
+++ b/Kitodo-XML-SchemaConverter/pom.xml
@@ -17,7 +17,7 @@
     <parent>
         <artifactId>kitodo-production</artifactId>
         <groupId>org.kitodo</groupId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - XML SchemaConverter</name>

--- a/Kitodo/pom.xml
+++ b/Kitodo/pom.xml
@@ -18,7 +18,7 @@
     <parent>
         <groupId>org.kitodo</groupId>
         <artifactId>kitodo-production</artifactId>
-        <version>${revision}</version>
+        <version>3.7.0-SNAPSHOT</version>
     </parent>
 
     <name>Kitodo - Core</name>

--- a/pom.xml
+++ b/pom.xml
@@ -19,7 +19,7 @@
 
     <groupId>org.kitodo</groupId>
     <artifactId>kitodo-production</artifactId>
-    <version>${revision}</version>
+    <version>3.7.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
     <organization>
@@ -30,8 +30,7 @@
     <url>https://github.com/kitodo/kitodo-production</url>
 
     <properties>
-        <revision>3.7.0-SNAPSHOT</revision>
-        <kitodo.version>${revision}</kitodo.version>
+        <kitodo.version>3.7.0-SNAPSHOT</kitodo.version>
         <checkstyle.config.location>config/checkstyle.xml</checkstyle.config.location>
         <main.basedir>${project.basedir}</main.basedir>
         <phase.prop>none</phase.prop>

